lib/mongo_mapper/plugins/dirty.rb

```diff
@@ -48,11 +52,18 @@ def write_key(key, value)
         else
           attribute_will_change!(key) unless attribute_changed?(key)
           super.tap do
-            changed_attributes.delete(key) unless attribute_value_changed?(key)
+            delete_changed_attributes(key) unless attribute_value_changed?(key)
           end
         end
       end
 
+      def delete_changed_attributes(key)
+        if ::ActiveModel::VERSION::MAJOR > 4
+          return clear_attribute_changes([key])
+        end
+        changed_attributes.delete(key)
```

Comment:
It seems we can use `changed_attributes([key])` even Rails 4.2

https://api.rubyonrails.org/v4.2.11/classes/ActiveModel/Dirty.html#method-i-clear_attribute_changesp

Reply:
tested, yes it can be used here :pray: thank you~!